spec/tapioca/compilers/dsl/active_resource_spec.rb

```diff
@@ -0,0 +1,207 @@
+# typed: false
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe("Tapioca::Compilers::Dsl::ActiveResource") do
+  before(:each) do
+    require "tapioca/compilers/dsl/active_resource"
+  end
+
+  subject do
+    Tapioca::Compilers::Dsl::ActiveResource.new
+  end
+
+  describe("#initialize") do
+    def constants_from(content)
+      with_content(content) do
+        subject.processable_constants.map(&:to_s).sort
+      end
+    end
+
+    it("gathers no constants if there are no ActiveResource classes") do
+      assert_empty(subject.processable_constants)
+    end
+
+    it("gathers only ActiveResource constants ") do
+      content = <<~RUBY
+        class Post < ActiveResource::Base
+        end
+
+        class Product < ActiveRecord::Base
```

Comment:
I guess you needed `active_record` for this to work, but it does not belong in the source file since it is only a testing concern.

Also, there is no specific reason why we need to test `ActiveRecord::Base` subclasses here. A nicer thing to test is that we are handling subclasses of subclasses of `ActiveResource::Base`, so:
```suggestion
        class Product < Post
```